File: sdks/csharp/src/PogodocApi/Render/Requests/InitializeRenderJobRequest.cs
using System.Text.Json.Serialization;
using PogodocApi;

#nullable enable

namespace PogodocApi;

public record InitializeRenderJobRequest
{
    [JsonPropertyName("data")]
    public Dictionary<string, object>? Data { get; init; }

    [JsonPropertyName("type")]
    public required InitializeRenderJobRequestType Type { get; init; }

    [JsonPropertyName("target")]
    public required InitializeRenderJobRequestTarget Target { get; init; }

    [JsonPropertyName("templateId")]
    public string? TemplateId { get; init; }

    [JsonPropertyName("formatOpts")]
    public InitializeRenderJobRequestFormatOpts? FormatOpts { get; init; }
}

File: sdks/php/sdk/src/Documents/Requests/InitializeRenderJobRequest.php
<?php

namespace Pogodoc\Documents\Requests;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Core\Json\JsonProperty;
use Pogodoc\Core\Types\ArrayType;
use Pogodoc\Documents\Types\InitializeRenderJobRequestType;
use Pogodoc\Documents\Types\InitializeRenderJobRequestTarget;
use Pogodoc\Documents\Types\InitializeRenderJobRequestFormatOpts;

class InitializeRenderJobRequest extends JsonSerializableType
{
    /**
     * @var ?array<string, mixed> $data
     */
    #[JsonProperty('data'), ArrayType(['string' => 'mixed'])]
    public ?array $data;

    /**
     * @var value-of<InitializeRenderJobRequestType> $type
     */
    #[JsonProperty('type')]
    public string $type;

    /**
     * @var value-of<InitializeRenderJobRequestTarget> $target
     */
    #[JsonProperty('target')]
    public string $target;

    /**
     * @var ?string $templateId
     */
    #[JsonProperty('templateId')]
    public ?string $templateId;

    /**
     * @var ?InitializeRenderJobRequestFormatOpts $formatOpts
     */
    #[JsonProperty('formatOpts')]
    public ?InitializeRenderJobRequestFormatOpts $formatOpts;

    /**
     * @param array{
     *   type: value-of<InitializeRenderJobRequestType>,
     *   target: value-of<InitializeRenderJobRequestTarget>,
     *   data?: ?array<string, mixed>,
     *   templateId?: ?string,
     *   formatOpts?: ?InitializeRenderJobRequestFormatOpts,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->data = $values['data'] ?? null;
        $this->type = $values['type'];
        $this->target = $values['target'];
        $this->templateId = $values['templateId'] ?? null;
        $this->formatOpts = $values['formatOpts'] ?? null;
    }
}
